extends Area3D

@export var page_index: int = 0  # Num√©ro de la page (0 √† 8)
@onready var audio_player: AudioStreamPlayer3D = $AudioStreamPlayer3D

func _ready():
	connect("body_entered", Callable(self, "_on_body_entered"))

func _on_body_entered(body):
	if not body.is_in_group("player"):
		return

	if page_index in Global.pages_trouvees:
		return

	Global.pages_trouvees.append(page_index)
	print("üìò Page trouv√©e :", page_index, "| Total :", Global.pages_trouvees.size(), "/ 9")

	play_random_sound()

	# V√©rifie si toutes les pages sont trouv√©es
	if Global.pages_trouvees.size() >= 9:
		await get_tree().create_timer(0.5).timeout  # petit d√©lai pour laisser le son se lancer
		get_tree().change_scene_to_file("res://UI/cinematique2.tscn")

	# Supprime la page apr√®s le son
	await get_tree().create_timer(1).timeout
	queue_free()

func play_random_sound():
	var random_id = randi_range(1, 4)
	var sound_path = "res://assets/audio/doublage/Trouv√©!" + ("" if random_id == 1 else str(random_id)) + ".wav"

	var sound = load(sound_path)
	if sound:
		audio_player.stream = sound
		audio_player.play()
	else:
		print("‚ö†Ô∏è Fichier introuvable :", sound_path)
